api/src/main/java/org/pmiops/workbench/billing/InitialCreditsService.java

```diff
@@ -242,6 +258,155 @@ public double getWorkspaceCreatorFreeCreditsRemaining(DbWorkspace dbWorkspace) {
     return Math.max(creatorFreeCreditsRemaining, 0);
   }
 
+  public void checkCreditsExpirationForUserIDs(List<Long> userIdsList) {
+    if (userIdsList != null && !userIdsList.isEmpty()) {
+      Iterable<DbUser> users = userDao.findAllById(userIdsList);
+      users.forEach(this::checkExpiration);
+    }
+  }
+
+  public Optional<Timestamp> getCreditsExpiration(DbUser user) {
```

Comment:
I think I left similar comments in one of previous PR. 
We should use java.time library - Instant here

`sql.Timestamp` is [bad API](https://dzone.com/articles/game-over-jdks-date-and-time-c). Even Oracle now [acknowledges](https://docs.oracle.com/javase/tutorial/datetime/iso/legacy.html) that these classes have "several drawbacks".

But I know Timestamp is retrieved from DbUser, and we can not change that. Not sure how much effort to add time conversion here. So it's upto you, fine to keep it as it if converting is hard in this class.

Reply:
Thanks for raising this. I'd prefer to address this at a later time given that the scope of this PR is already quite large.